spec.html

```diff
@@ -27388,10 +27388,15 @@ <h1>Overview of Date Objects and Definitions of Abstract Operations</h1>
       <!-- es6num="20.3.1.1" -->
       <emu-clause id="sec-time-values-and-time-range">
         <h1>Time Values and Time Range</h1>
-        <p>A Date object contains a Number indicating a particular instant in time to within a millisecond. Such a Number is called a <dfn>time value</dfn>. A time value may also be *NaN*, indicating that the Date object does not represent a specific instant of time.</p>
-        <p>Time is measured in ECMAScript in milliseconds since 01 January, 1970 UTC. In time values leap seconds are ignored. It is assumed that there are exactly 86,400,000 milliseconds per day. ECMAScript Number values can represent all integers from -9,007,199,254,740,992 to 9,007,199,254,740,992; this range suffices to measure times to millisecond precision for any instant that is within approximately 285,616 years, either forward or backward, from 01 January, 1970 UTC.</p>
-        <p>The actual range of times supported by ECMAScript Date objects is slightly smaller: exactly -100,000,000 days to 100,000,000 days measured relative to midnight at the beginning of 01 January, 1970 UTC. This gives a range of 8,640,000,000,000,000 milliseconds to either side of 01 January, 1970 UTC.</p>
-        <p>The exact moment of midnight at the beginning of 01 January, 1970 UTC is represented by the value *+0*.</p>
+        <p>A Date object contains a Number representing an instant in time with millisecond precision. Such a Number is called a <dfn>time value</dfn>. A time value may also be *NaN*, indicating that the Date object does not represent a specific instant in time.</p>
+        <p>Time is measured in ECMAScript as milliseconds since midnight at the beginning of 01 January, 1970 UTC. Time in ECMAScript does not observe leap seconds; they are ignored. Time calculations assume a day contains exactly <emu-eqn>60 &times; 60 &times; 24 &times; 1000 = 86,400,000</emu-eqn> milliseconds.</p>
```

Comment:
If we are going this deep, do we want to discuss *why* we don't support leap seconds? I feel like that gets asked a lot, and of course it goes back to the fact that we're fundamentally using Unix time which doesn't support them here.

Small nit - I believe we should say `time calculations assume a UTC day contains exactly...`

Reply:
>  I believe we should say `time calculations assume a UTC day contains exactly...`

(I.e., inserting "UTC".) I think we should go in the opposite direction, removing most mentions of UTC, for the very reason that ES doesn't support leap seconds.

Reply:
@maggiepint I can ask around tomorrow, but there might be some reason why Unix / POSIX can't be referenced specifically in the spec (neither are currently).

@jmdyck I'd tend to agree re: leaving UTC out when referring to the length of a day. One of the most common references for people (Wikipedia) clearly states a UTC day sometimes contains an extra second, so it would be like the spec saying "A UTC day is not a UTC day".

As for removing most mentions of UTC, the spec currently has 69 lines that mention UTC at least once. Even if removal is technically sound, careful wording would have to replace it and it's a larger proposal that would have to be discussed separately.

Reply:
> As for removing most mentions of UTC, [...] it's a larger proposal that would have to be discussed separately.

Agreed.

Reply:
I suppose UTC isn't right.

What I'm going for is *not local*, because of course the spec does support
local time which, due to DST or zone splitting, has days not of that length.

Reply:
I asked @bterlson and he didn't see an issue with mentioning the alignment with the POSIX specification of a day, so see my last push for proposed wording.

@jmdyck When looking back over the POSIX spec, it appears they did exactly as you are suggesting and changed all references from "**seconds since 00:00:00 UTC (Coordinated
Universal Time), January 1 1970**" to "**seconds since the Epoch**". I'd be ok with attempting to tackle that as a subsequent PR. I'm guessing that might need to be brought up to the committee because it would likely involve defining the ES Epoch explicitly as 1970-01-01T00:00:00 TAI and adding similar implementation-defined language that POSIX resorted to:

> The relationship between the actual time of day and the current value for seconds since the Epoch is unspecified.

> How any changes to the value of seconds since the Epoch are made to align to a desired relationship with the current actual time is implementation-defined. As represented in seconds since the Epoch, each and every day shall be accounted for by exactly 86400 seconds.